upgrade/upgrade-4.0.0.php

```diff
@@ -0,0 +1,162 @@
+        `id_psreassurance` int(10) unsigned NOT NULL,
+        `id_lang` int(10) unsigned NOT NULL,
+        `id_shop` int(10) unsigned NOT NULL,
+        `title` varchar(255) NOT NULL,
+        `description` varchar(255) NOT NULL,
+        `link` varchar(255) NOT NULL,
+        PRIMARY KEY (`id_psreassurance`,`id_shop`,`id_lang`)
+        ) ENGINE='._MYSQL_ENGINE_.' DEFAULT CHARSET=UTF8;';
+
+    /*
+    ** First we make a verification that the new tables are empty
+    */
+    $sql[] = 'TRUNCATE TABLE `'._DB_PREFIX_.'psreassurance`';
+    $sql[] = 'TRUNCATE TABLE `'._DB_PREFIX_.'psreassurance_lang`';
+
+    /* This path : /modules/blockreassurance/img/".$reassurance['file_name']
+    ** is used with the real path in the module
+    **
+    ** We do the INSERT INTO to get the old module values
+    */
+    if (!empty($reassurances)) {
+        foreach ($reassurances as $reassurance) {
+            $sql[] = "INSERT INTO "._DB_PREFIX_."psreassurance (id_psreassurance, icone, icone_perso, status, position, id_shop, type_link, id_cms, date_add)
+                VALUES (".$reassurance['id_reassurance'].", '".$module->old_path_img.$reassurance['file_name']."', null, 1, ".$reassurance['id_reassurance'].", ".$reassurance['id_shop'].", null, null, now())";
+        }
+    }
+
+    if (!empty($reassurance_langs)) {
+        foreach ($reassurance_langs as $reassurance_lang) {
+            $sql[] = "INSERT INTO "._DB_PREFIX_."psreassurance_lang (id_psreassurance, id_lang, id_shop, title, description, link)
+            VALUES (".$reassurance_lang['id_reassurance'].", ".$reassurance_lang['id_lang'].", 1, '".$reassurance_lang['text']."', '', '')";
+        }
+    }
+
+    /*
+    ** Here we execute the SQL
+    */
+    foreach ($sql as $query) {
+        if (Db::getInstance()->execute($query) == false) {
+            return Db::getInstance()->getMsgError();
+        }
+    }
+
+    $result = true;
+    /*
+    ** Verification if the hooks are already registered
+    */
+    if (!$module->isRegisteredInHook('displayAfterBodyOpeningTag')) {
+        $result &= $module->registerHook('displayAfterBodyOpeningTag');
+    }
+
+    if (!$module->isRegisteredInHook('displayNavFullWidth')) {
+        $result &= $module->registerHook('displayNavFullWidth');
+    }
+
+    if (!$module->isRegisteredInHook('displayFooterAfter')) {
+        $result &= $module->registerHook('displayFooterAfter');
+    }
+
+    if (!$module->isRegisteredInHook('displayFooterBefore')) {
+        $result &= $module->registerHook('displayFooterBefore');
+    }
+
+    if (!$module->isRegisteredInHook('displayReassurance')) {
+        $result &= $module->registerHook('displayReassurance');
+    }
+
+    if (!$module->isRegisteredInHook('actionFrontControllerSetMedia')) {
+        $result &= $module->registerHook('actionFrontControllerSetMedia');
+    }
+    /*
+    ** We set the new configuration values
+    */
+    Configuration::updateValue('PSR_HOOK_HEADER', '0');
```

Comment:
Here's an idea, but not mandatory: if we could put these statements `Configuration::updateValue(...)` in a common php class (like "ReassuranceInstallUtils.php" for example) then we could avoid to duplicate them.

Reply:
ok